ballerina-tests/http-advanced-tests/tests/relaxed_data_binding_tests.bal

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2024 WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+import ballerina/http;
+
+type Citizen record {
+    int id;
+    string name;
+    string address?;
+};
+
+type Student record {|
+    string name;
+    string module;
+    string grade?;
+|};
+
+type Teacher record {
+    int id;
+    string name;
+    string? subject;
+};
+
+listener http:Listener laxDataBindingEP = new (laxDataBindingTestPort, httpVersion = http:HTTP_2_0);
+final http:Client laxClient = check new ("http://localhost:" + laxDataBindingTestPort.toString(), {
```

Comment:
Lets have negative test cases with normal HTTP client where all these special relaxed scenarios should fail

Reply:
Added negative tests with this commit: [f2c9eab9854ad8dc310c730d1a2909e42f1780d1](https://github.com/ballerina-platform/module-ballerina-http/pull/2259/commits/f2c9eab9854ad8dc310c730d1a2909e42f1780d1)